internal/otelcollector/config/tracegateway/config_builder.go

```diff
@@ -32,94 +32,284 @@ type BuildOptions struct {
 }
 
 func (b *Builder) Build(ctx context.Context, pipelines []telemetryv1alpha1.TracePipeline, opts BuildOptions) (*Config, common.EnvVars, error) {
-	b.config = b.baseConfig(opts)
+	b.config = &Config{
+		Base:       common.BaseConfig(),
+		Receivers:  make(map[string]any),
+		Processors: make(map[string]any),
+		Exporters:  make(map[string]any),
+	}
 	b.envVars = make(common.EnvVars)
 
 	// Iterate over each TracePipeline CR and enrich the config with pipeline-specific components
 	queueSize := maxQueueSize / len(pipelines)
 
 	for i := range pipelines {
-		pipeline := pipelines[i]
-
-		if err := b.addComponentsForTracePipeline(ctx, &pipeline, queueSize); err != nil {
-			return nil, nil, err
+		if err := b.addServicePipeline(ctx, &pipelines[i],
+			b.addOTLPReceiver(),
+			b.addMemoryLimiterProcessor(),
+			b.addK8sAttributesProcessor(opts),
+			b.addIstioNoiseFilterProcessor(),
+			b.addInsertClusterAttributesProcessor(opts),
+			b.addServiceEnrichmentProcessor(),
+			b.addDropKymaAttributesProcessor(),
+			b.addUserDefinedTransformProcessor(),
+			b.addBatchProcessor(),
+			b.addOTLPExporter(queueSize),
+		); err != nil {
+			return nil, nil, fmt.Errorf("failed to add service pipeline: %w", err)
 		}
-
-		b.addServicePipelines(&pipeline)
 	}
 
 	return b.config, b.envVars, nil
 }
 
-// baseConfig creates the static/global base configuration for the trace gateway collector.
-// Pipeline-specific components are added later via addComponentsForTracePipeline method.
-func (b *Builder) baseConfig(opts BuildOptions) *Config {
-	return &Config{
-		Base: common.Base{
-			Service:    common.ServiceConfig(make(common.Pipelines)),
-			Extensions: common.ExtensionsConfig(),
-		},
-		Receivers:  receiversConfig(),
-		Processors: processorsConfig(opts),
-		Exporters:  make(Exporters),
+// buildComponentFunc defines a function type for building components in the telemetry configuration
+type buildComponentFunc = func(ctx context.Context, tp *telemetryv1alpha1.TracePipeline) error
```

Comment:
Is there a reason to have the `buildComponentFunc` as a type alias(with an `=`) instead of a type definition (without a `=`)?